### PR TITLE
Feature/e UI 3818 duration access

### PR DESCRIPTION
**Pull request by udaydenduluri33**
https://tools.hmcts.net/jira/browse/EUI-3818
https://tools.hmcts.net/jira/browse/EUI-3819
https://tools.hmcts.net/jira/browse/EUI-3961
https://tools.hmcts.net/jira/browse/EUI-3864
https://tools.hmcts.net/jira/browse/EUI-3866

### Change description ###

Display User story - Roles and Access Tab

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by connorpgpmcelroy — APPROVED** [review]
All looks good to me. Could add test based on display of radio buttons but do not think it's necessary

File: src/role-access/containers/allocate-role/choose-duration/choose-duration.component.ts [hunk 1]
```diff
@@ -1,28 +1,199 @@
...
+    const currentDate = new Date();
+    currentDate.setHours(0, 0, 0, 0);
+    return currentDate;
+  }
+
+  public getStartDate() {
+    return new Date(this.yearStartDate.value, this.monthStartDate.value - 1, this.dayStartDate.value);
+  }
+
+  public getEndDate() {
```
Comment (kuda1992) [inline src/role-access/containers/allocate-role/choose-duration/choose-duration.component.ts:185]: **Minor**  missing return type

File: src/role-access/containers/allocate-role/choose-duration/choose-duration.component.ts [hunk 2]
```diff
@@ -1,28 +1,199 @@
...
+    return dateMissing;
+  }
+
+  public getTodayDate() {
+    const currentDate = new Date();
+    currentDate.setHours(0, 0, 0, 0);
+    return currentDate;
+  }
+
+  public getStartDate() {
```
Comment (kuda1992) [inline src/role-access/containers/allocate-role/choose-duration/choose-duration.component.ts:181]: **Minor**  missing return type

File: src/role-access/containers/allocate-role/choose-duration/choose-duration.component.ts [hunk 3]
```diff
@@ -1,28 +1,199 @@
...
+
+    if (!parseInt(this.dayEndDate.value, 10) || !parseInt(this.monthEndDate.value, 10) || !parseInt(this.yearEndDate.value, 10)) {
+      this.endDateErrorMessage = 'Please enter some value';
+      this.isEndDateError = true;
+      dateMissing = false;
+    }
+    return dateMissing;
+  }
+
+  public getTodayDate() {
```
Comment (kuda1992) [inline src/role-access/containers/allocate-role/choose-duration/choose-duration.component.ts:175]: **Minor**  missing return type

File: src/role-access/containers/allocate-role/choose-duration/choose-duration.component.ts [hunk 4]
```diff
@@ -1,28 +1,199 @@
...
+
+  public getStartDate() {
+    return new Date(this.yearStartDate.value, this.monthStartDate.value - 1, this.dayStartDate.value);
+  }
+
+  public getEndDate() {
+    return new Date(this.yearEndDate.value, this.monthEndDate.value - 1, this.dayEndDate.value);
+  }
+
+  public onItemChange(item: DurationOfRole) {
```
Comment (kuda1992) [inline src/role-access/containers/allocate-role/choose-duration/choose-duration.component.ts:189]: **Minor**  missing return type

**Review by kuda1992 — CHANGES_REQUESTED** [review]
All good apart from little minor changes

File: src/role-access/containers/allocate-role/choose-duration/choose-duration.component.ts (hunk 4 above)
Comment (udaydenduluri33) [reply src/role-access/containers/allocate-role/choose-duration/choose-duration.component.ts:189]: Done

File: src/role-access/containers/allocate-role/choose-duration/choose-duration.component.ts (hunk 3 above)
Comment (udaydenduluri33) [reply src/role-access/containers/allocate-role/choose-duration/choose-duration.component.ts]: Done

File: src/role-access/containers/allocate-role/choose-duration/choose-duration.component.ts (hunk 2 above)
Comment (udaydenduluri33) [reply src/role-access/containers/allocate-role/choose-duration/choose-duration.component.ts]: Done

**Review by kuda1992 — APPROVED** [review]
All good

**Pull request merged**